ctera/src/main/java/ch/cyberduck/core/ctera/CteraAclPermissionFeature.java

```diff
@@ -0,0 +1,208 @@
+    public static final Acl.Role READPERMISSION = new Acl.Role("readpermission");
+
+    /**
+     * Write Data: Allows or denies making changes to a file and overwriting existing content.
+     */
+    // TODO CTERA-136 files only or what is the interplay with Createfilepermission/CreateDirectoriespermission?
+    public static final Acl.Role WRITEPERMISSION = new Acl.Role("writepermission");
+
+    /**
+     * Execute File: Allows or denies running program (executable) files.
+     * Files only.
+     */
+    public static final Acl.Role EXECUTEPERMISSION = new Acl.Role("executepermission");
+
+    /**
+     * Allows or denies deleting the file or folder. If you don't have Delete permission on a file or folder,
+     * you can still delete it if you have been granted Delete Subfolders and Files on the parent folder.
+     */
+    public static final Acl.Role DELETEPERMISSION = new Acl.Role("deletepermission");
+
+    /**
+     * Traverse Folder: Allows or denies moving through a restricted folder to reach files and folders
+     * beneath the restricted folder in the folder hierarchy. Traverse folder takes effect only when the group or user
+     * is not granted the "Bypass traverse checking user" right in the Group Policy snap-in.
+     * This permission does not automatically allow running program files.
+     * Directories only.
+     */
+    public static final Acl.Role TRAVERSEPERMISSION = new Acl.Role("traversepermission");
+
+    /**
+     * Create Files: Allows or denies creating files within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEFILEPERMISSION = new Acl.Role("Createfilepermission");
+
+    /**
+     * Create Folders: Allows or denies creating subfolders within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEDIRECTORIESPERMISSION = new Acl.Role("CreateDirectoriespermission");
+
+    static final List<Acl.Role> allCteraCustomACLRoles = Collections.unmodifiableList(Arrays.asList(
+            READPERMISSION, WRITEPERMISSION, EXECUTEPERMISSION, DELETEPERMISSION, TRAVERSEPERMISSION, CREATEFILEPERMISSION, CREATEDIRECTORIESPERMISSION
+    ));
+
+    final static List<QName> allCteraCustomACLQn = Collections.unmodifiableList(allCteraCustomACLRoles.stream().map(CteraAclPermissionFeature::toQn).collect(Collectors.toList()));
+
+    public static QName toQn(final Acl.Role role) {
+        return new QName(CTERA_NAMESPACE_URI, role.getName(), CTERA_NAMESPACE_PREFIX);
+    }
+
+    public static String toProp(final QName qn) {
+        return qn.getLocalPart();
+    }
+
+    public static Acl.Role toRole(final QName qn) {
+        return new Acl.Role(toProp(qn));
+    }
+
+    public static Acl customPropsToAcl(final Map<String, String> customProps) {
+        if(customProps.isEmpty()) {
+            return Acl.EMPTY;
+        }
```

Comment:
As you previously commented this should be omitted because custom properties may contain other user metadata?

Reply:
We iterate over the expected custom properties and add the respective roles, so lgtm. 
Or does "no custom properties" mean "some default set of roles" (700?) and not no roles (as implemented)? 

However, we have an asymmetry wrt `aclToPermission`, where do "empty acl" to `700`:
```
if(Acl.EMPTY.equals(acl)) {
        return new Permission(700);
    }
```

Reply:
Discussed offline: add `preflight` tests ensuring backwards compatibility.